### PR TITLE
DBZ-8114 Implementing VariableScaleDecimal and improving tests

### PR DESCRIPTION
**Pull request by nrkljo**

### CODE REVIEW

File: src/main/java/io/debezium/connector/informix/InformixValueConverters.java [hunk 1]
```diff
@@ -41,20 +51,106 @@ public InformixValueConverters(DecimalMode decimalMode, TemporalPrecisionMode te
...
+                SchemaBuilder builder = super.schemaBuilder(column);
+                logger.debug("JdbcValueConverters returned '{}' for column '{}'", builder != null ? builder.getClass().getName() : null, column.name());
+                return builder;
+        }
+    }
+
+    private SchemaBuilder getNumericSchema(Column column) {
+        if (column.scale().isPresent()) {
+
+            if (column.scale().get() == 255 && decimalMode == DecimalMode.PRECISE) {
```
Comment (Naros) [inline src/main/java/io/debezium/connector/informix/InformixValueConverters.java:75]: Could we move the `255` to a private static final class variable with a descriptive name to clarify precisely what 255 means?  The value itself is less important than its purpose.  There is another instance of this in a method below around like ~108.

File: src/main/java/io/debezium/connector/informix/InformixValueConverters.java [hunk 2]
```diff
@@ -41,20 +51,106 @@ public InformixValueConverters(DecimalMode decimalMode, TemporalPrecisionMode te
...
+                    receiver.deliver(new BufferedReader(((IfxCblob) data).getCharacterStream()).lines().collect(Collectors.joining(System.lineSeparator())));
+                }
+                catch (SQLException e) {
+                    throw new RuntimeException("Error processing data from " + column.jdbcType() + " and column " + column +
+                            ": class=" + data.getClass(), e);
+                }
+            });
+        }
+        return super.convertString(column, fieldDefn, data);
+    }
```
Comment (Naros) [inline src/main/java/io/debezium/connector/informix/InformixValueConverters.java:152]: I'm assuming this change is specifically for tests?  
It just caught my eye as it seems unrelated to `VariableScaleDecimal` behavior, no?

**Review by Naros — COMMENTED** [review]
Just two small in-line questions/nit-picks.

File: src/main/java/io/debezium/connector/informix/InformixValueConverters.java (hunk 2 above)
Comment (nrkljo) [reply src/main/java/io/debezium/connector/informix/InformixValueConverters.java:152]: Yes, I adapted the AbstractDatatypesTest from the Oracle-connector, and the CLOB needed some care

File: src/main/java/io/debezium/connector/informix/InformixValueConverters.java (hunk 1 above)
Comment (nrkljo) [reply src/main/java/io/debezium/connector/informix/InformixValueConverters.java:75]: Certainly. 
I'm unsure if Informix actually encodes scale as a signed byte when everyone else thinks it is a wider field, and sends -1 to signal variable scale or if this is just an arbitrary number. I noticed that DataGrip "successfully" interpret these fields has having an actual scale of 255...
